src/s3ql/backends/gdrive.py

```diff
@@ -0,0 +1,599 @@
+
+    def _lookup_file(self, name, folder=None):
+        '''Lookup file by name
+        Arguments:
+        name -- file name. May contain absolute or relative file name with '/' as path separator
+        folder -- folder to start lookup from (root folder if omitted)
+        '''
+        if folder is None:
+            folder = { 'id': 'root', 'path': '/', 'mimeType': Backend.MIME_TYPE_FOLDER }
+
+        # find next path component
+        if len(name) == 0:
+            return folder
+        if name[0] == '/':
+            name = name[1:]
+        n = name.split('/', 1)
+        if len(n[0]) == 0:
+            return folder
+
+        # iterate over folder children
+        query = "name = '{0}'".format(self._escape_string(n[0]))
+        for f in self._list_files(folder, "id, name, mimeType, size,md5Checksum, properties", query):
+            f['path'] = folder['path'] + f['name']
+            if f['mimeType'] == Backend.MIME_TYPE_FOLDER:
+                f['path'] += '/'
+
+            log.debug("{0} ({1}: {2})".format(f['path'], f['id'], f['mimeType']))
+            if len(n) == 1:
+                return f
+            elif f['mimeType'] == Backend.MIME_TYPE_FOLDER:
+                return self._lookup_file(n[1], f)
+            else:
+                return None
+
+        # nothing found
+        return None
+
+    @staticmethod
+    def _chunk_property(k, i):
+        '''Make chunk property'''
+        return 'ref({0}.{1})'.format(k, i)
+
+    def _encode_metadata(self, metadata):
+        #
+        # Google Drive properties are limited to 124 bytes for key+value
+        #
+        # 'bytes' metadata properties are base64-encoded and are assumed
+        # the only properties that may not fit 124 bytes
+        #
+        # if 'bytes' property fits it is stored as 'b64:' + <base64 value>
+        # otherwise it is stored as 'ref:' + <number of chunks>
+        #
+        # 'ref' data is stored in chunks (of MAX_REF_CHUNK_SIZE) in properties
+        # with name 'ref(name.chunk-index)'
+        if metadata is None:
+            return None
+        properties = dict()
+        for k, v in metadata.items():
+            if isinstance(v, bytes):
+                b64 = base64.b64encode(v).decode()
+                value = Backend.VALUE_BASE64 + b64
+                if len(k+value) <= Backend.MAX_PROPERTY_SIZE:
+                    properties[k] = value
+                else:
+                    # create chunked metadata property
+                    chunks = [b64[i:i+Backend.MAX_REF_CHUNK_SIZE]
+                        for i in range(0, len(b64), Backend.MAX_REF_CHUNK_SIZE)]
+                    properties[k] = Backend.VALUE_REF + str(len(chunks))
+                    for i, chunk in enumerate(chunks):
+                        properties[self._chunk_property(k, i)] = chunk
+            else:
+                properties[k] = v
+        return properties
+
+    def _decode_metadata(self, f):
```

Comment:
This looks like you really should be using `thaw_basic_mapping()` and `free_basic_mapping` instead (from `s3ql.common`)

Reply:
I will check it, I didn't know about this